src/main/java/org/simulator/io/CSVImporter.java

```diff
@@ -54,19 +48,63 @@
   private static final transient Logger logger = Logger.getLogger(CSVImporter.class.getName());
 
   /**
-   * The header of the file
+   * This method reads the data from the CSV file and converts it
+   * into the map with key as the column name and value as the array of amounts
+   * for the particular column.
+   * 
+   * @param pathname The path of the CSV file
+   * @return columnsMap -> LinkedListHashMap
    */
-  private String[] header;
+  private Map<String, double[]> readDataFromCSV(String pathname) throws IOException {
+    Map<String, double[]> columnsMap = new LinkedHashMap<>();
+    BufferedReader reader = new BufferedReader(new FileReader(pathname));
+    String line = reader.readLine();
+    List<String> lines = new LinkedList<String>();
+    String[] identifiers = line.split(",");
+    if (line != null) {
+      line = reader.readLine();
+      while ((line != null) && !line.isEmpty()) {
+        lines.add(line);
+        line = reader.readLine();
+      }
+
+      if (identifiers[0].equalsIgnoreCase("time")){
+        identifiers[0] = identifiers[0].toLowerCase();
+      }
+      for (String s : identifiers) {
+        columnsMap.put(s, new double[lines.size()]);
+      }
+
+      for (int i = 0; i < lines.size(); i++) {
+        String[] column = lines.get(i).split(",");
+        for (int j = 0; j < column.length; j++) {
+          if ((column[j] != null) && (column[j].length() > 0)) {
+            if (column[j].equalsIgnoreCase("INF")) {
+              columnsMap.get(identifiers[j])[i] = Double.POSITIVE_INFINITY;
+            } else if (column[j].equalsIgnoreCase("-INF")) {
+              columnsMap.get(identifiers[j])[i] = Double.NEGATIVE_INFINITY;
+            } else if (column[j].equalsIgnoreCase("NAN")) {
+              columnsMap.get(identifiers[j])[i] = Double.NaN;
```

Comment:
Better use the existing method [Double.parsedouble](https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/lang/Double.html#parseDouble(java.lang.String)) from Java.

Reply:
For string "INF" and "-INF", it will return NumberFormatException. Yeah, but I have updated for the NaN case.